### ISSUE TITLE
test_fcs_stream fail with i40evf interface

### ISSUE THREAD

**Issue by jsmoon** [cisco-system-traffic-generator/trex-core#737]
I performed STL regression with i40evf interface and got the following error result.

```
1) ERROR: test_fcs_stream (stateless_tests.stl_rx_test.STLRX_Test)

   Traceback (most recent call last):
    stateless_tests/stl_rx_test.py line 233 in wrapped
      func(self, *args, **kwargs)
    stateless_tests/stl_rx_test.py line 870 in test_fcs_stream
      self.send_1_burst(ports, lat, 100)
    stateless_tests/stl_rx_test.py line 858 in send_1_burst
      raise STLError(msg)
   TRexError: Stats do not match the expected:
   ERROR field : tps[ibytes], read : 12000 != expected : 12800 
   ERROR field : stats[1][ibytes], read : 6000 != expected : 6400 
   ERROR field : stats[0][ibytes], read : 6000 != expected : 6400 
```

@hhaim have you got this issue?

**Comment by hhaim** [comment]
@jsmoon we don't see this issue on our setups maybe it is related to PF driver (linux) or firmware

**Comment by jsmoon** [comment]
@hhaim could I get your PF driver and firmware information?

```
# ethtool -i enp59s0f0
driver: i40e
version: 2.1.14-k
firmware-version: 6.80 0x80003d1a 18.8.9
expansion-rom-version: 
bus-info: 0000:3b:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: yes
```
```
# modinfo i40e
filename:       /lib/modules/4.15.0-46-generic/kernel/drivers/net/ethernet/intel/i40e/i40e.ko
version:        2.1.14-k
license:        GPL
description:    Intel(R) Ethernet Connection XL710 Network Driver
author:         Intel Corporation, <e1000-devel@lists.sourceforge.net>
srcversion:     8BFD7F714CCDCD64E49C5DB
alias:          pci:v00008086d0000158Bsv*sd*bc*sc*i*
alias:          pci:v00008086d0000158Asv*sd*bc*sc*i*
alias:          pci:v00008086d00001588sv*sd*bc*sc*i*
alias:          pci:v00008086d00001587sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D3sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D2sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D1sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D0sv*sd*bc*sc*i*
alias:          pci:v00008086d000037CFsv*sd*bc*sc*i*
alias:          pci:v00008086d000037CEsv*sd*bc*sc*i*
alias:          pci:v00008086d00001589sv*sd*bc*sc*i*
alias:          pci:v00008086d00001586sv*sd*bc*sc*i*
alias:          pci:v00008086d00001585sv*sd*bc*sc*i*
alias:          pci:v00008086d00001584sv*sd*bc*sc*i*
alias:          pci:v00008086d00001583sv*sd*bc*sc*i*
alias:          pci:v00008086d00001581sv*sd*bc*sc*i*
alias:          pci:v00008086d00001580sv*sd*bc*sc*i*
alias:          pci:v00008086d00001574sv*sd*bc*sc*i*
alias:          pci:v00008086d00001572sv*sd*bc*sc*i*
depends:        ptp
retpoline:      Y
intree:         Y
name:           i40e
vermagic:       4.15.0-46-generic SMP mod_unload 
signat:         PKCS#7
signer:         
sig_key:        
sig_hashalgo:   md4
parm:           debug:Debug level (0=none,...,16=all), Debug mask (0x8XXXXXXX) (uint)
```

**Comment by hhaim** [comment]
```
[]# sudo ethtool -i enp5s0f0
driver: i40e
version: 2.3.6
firmware-version: 5.04 0x800024d0 0.0.0
expansion-rom-version: 
bus-info: 0000:05:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: yes


filename:       /lib/modules/3.10.0-693.el7.x86_64/updates/drivers/net/ethernet/intel/i40e/i40e.ko
version:        2.3.6
license:        GPL
description:    Intel(R) 40-10 Gigabit Ethernet Connection Network Driver
author:         Intel Corporation, <e1000-devel@lists.sourceforge.net>
rhelversion:    7.4
srcversion:     934AB35A0D5F3984E6B4C9E
alias:          pci:v00008086d0000158Bsv*sd*bc*sc*i*
alias:          pci:v00008086d0000158Asv*sd*bc*sc*i*
alias:          pci:v00008086d000037D3sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D2sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D1sv*sd*bc*sc*i*
alias:          pci:v00008086d000037D0sv*sd*bc*sc*i*
alias:          pci:v00008086d000037CFsv*sd*bc*sc*i*
alias:          pci:v00008086d000037CEsv*sd*bc*sc*i*
alias:          pci:v00008086d00001588sv*sd*bc*sc*i*
alias:          pci:v00008086d00001587sv*sd*bc*sc*i*
alias:          pci:v00008086d00001589sv*sd*bc*sc*i*
alias:          pci:v00008086d00001586sv*sd*bc*sc*i*
alias:          pci:v00008086d00001585sv*sd*bc*sc*i*
alias:          pci:v00008086d00001584sv*sd*bc*sc*i*
alias:          pci:v00008086d00001583sv*sd*bc*sc*i*
alias:          pci:v00008086d00001581sv*sd*bc*sc*i*
alias:          pci:v00008086d00001580sv*sd*bc*sc*i*
alias:          pci:v00008086d00001574sv*sd*bc*sc*i*
alias:          pci:v00008086d00001572sv*sd*bc*sc*i*
depends:        ptp
vermagic:       3.10.0-693.el7.x86_64 SMP mod_unload modversions 
parm:           debug:Debug level (0=none,...,16=all) (int)

```

**Comment by jsmoon** [comment]
@hhaim I have tried it with a new i40e driver. But the result is the same.

```
# ethtool -i enp59s0f0
driver: i40e
version: 2.15.9
firmware-version: 6.80 0x80003d1a 18.8.9
expansion-rom-version: 
bus-info: 0000:3b:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: yes
```

I think `trex_22` may not run the `test_fcs_stream` test case.
```
 170         self.i40e_vf_defect_github_200 = CTRexScenario.setup_name in ['trex22']

 861     @try_few_times_on_vm
 862     def test_fcs_stream(self):
 863         if self.i40e_vf_defect_github_200:
 864             self.skip('Skip for i40e_vf setup')
```

Could you please check this issue manually on your side?

**Comment by hhaim** [comment]
@jsmoon you are right, this is an old issue #200 is the same as this one and @bdollma already mask it.

**Comment by jsmoon** [comment]
@hhaim, but in my experience, this issue was not observed before DPDK 21.02 merged.
I have found the following change (805be3b0 at src/dpdk/drivers/net/i40e/i40e_ethdev_vf.c) merged.
```
@@ -2357,6 +2434,7 @@ i40evf_dev_stats_get(struct rte_eth_dev *dev, struct rte_eth_stats *stats)
                stats->imissed = pstats->rx_discards;
                stats->oerrors = pstats->tx_errors + pstats->tx_discards;
                stats->ibytes = pstats->rx_bytes;
+               stats->ibytes -= stats->ipackets * RTE_ETHER_CRC_LEN;
                stats->obytes = pstats->tx_bytes;
 #ifdef TREX_PATCH
     }
```
What do you think about this?

**Comment by besdollma** [comment]
@hhaim , @jsmoon 
Issue 200 was opened only so we could reference it in code, there is no commit to fix it.

[Reference](https://github.com/cisco-system-traffic-generator/trex-core/blob/ad9f32bc7bb9d23a2175ee5f358dc61e8b832adf/scripts/automation/regression/stateless_tests/stl_rx_test.py#L164)

[Commit](https://github.com/cisco-system-traffic-generator/trex-core/commit/ad9f32bc7bb9d23a2175ee5f358dc61e8b832adf?branch=ad9f32bc7bb9d23a2175ee5f358dc61e8b832adf&diff=split)

Thanks,

**Comment by hhaim** [comment]
@jsmoon from the code you sent it seems it is possible to fix the original issue #200
